### PR TITLE
MUIC-610: [Android] - Make Positive button and Negative button styles configurable

### PR DESCRIPTION
**Pull request by deniss-mc**
# https://glia.atlassian.net/browse/MUIC-610

Notes:
Examples added for runtime and buildtime configuration

### CODE REVIEW

**Review by dukhovnyi — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/view/button/BaseConfigurableButton.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                .textConfiguration(textConfiguration)
+                .backgroundColor(getBackgroundTintList())
+                .strokeColor(getStrokeColor())
+                .strokeWidth(getStrokeWidth())
+                .build();
+    }
+
+    public void setTheme(UiTheme theme) {
+        ButtonConfiguration runTimeConfiguration = getButtonConfigurationFromTheme(theme);
+        if (theme == null || runTimeConfiguration == null) return;
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/view/button/BaseConfigurableButton.java:81]: why theme nullcheck if it's not used further?

**Review by jakob-mc — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/themes.xml [hunk 2]
```diff
@@ -35,7 +44,25 @@
         <item name="android:backgroundTint">@color/color_dark_cyan</item>
         <item name="android:textColor">@color/color_pure_yellow</item>
         <item name="android:strokeColor">?attr/gliaSystemNegativeColor</item>
-        <item name="android:fontFamily">@font/tangerine</item>
+        <!-- item name="android:fontFamily">@font/tangerine</item -->
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply app/src/main/res/values/themes.xml:47]: Its example

File: app/src/main/java/com/glia/exampleapp/Utils.java [hunk 3]
```diff
@@ -57,10 +62,47 @@ public static UiTheme getUiThemeByPrefs(SharedPreferences sharedPreferences, Res
...
+        builder.setNeutralButtonConfiguration(null);
 
         return builder.build();
     }
 
+    // Positive button configuration Example runtime
+    private static ButtonConfiguration getPositiveButtonTestingConfiguration(Context context) {
+        return ButtonConfiguration.builder()
+                .backgroundColor(
+                        ColorStateList.valueOf(ContextCompat.getColor(context, R.color.color_blue))
```
[the remark this answers is not in the document]
Comment (deniss-mc) [reply app/src/main/java/com/glia/exampleapp/Utils.java:76]: check the function name - no helper methods is required as it is builder usage example

[the remark this answers is not in the document]
Comment (deniss-mc) [reply app/src/main/java/com/glia/exampleapp/Utils.java:76]: function is not used anywhere

File: widgetssdk/src/main/java/com/glia/widgets/view/button/BaseConfigurableButton.java (hunk 1 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/view/button/BaseConfigurableButton.java:81]: good catch

File: app/src/main/res/values/themes.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakob-mc) [reply app/src/main/res/values/themes.xml:47]: then a little comment wouldn't hurt. otherwise it looks like was left in unintentionally.

**Review by jakob-mc — APPROVED** [review]

**Pull request merged**